flocx_market/api/offer_contract_relationship.py

```diff
@@ -6,41 +6,59 @@
 class OfferContractRelationship(Resource):
 
     @classmethod
-    def get(cls, marketplace_offer_id=None, contract_id=None):
-        if (contract_id is None) and (marketplace_offer_id is None):
+    def get(cls, offer_contract_relationship_id=None):
+        if offer_contract_relationship_id:
+            ocr_with_id = ocr.OfferContractRelationship\
+                .get(g.context, offer_contract_relationship_id)
+            if ocr_with_id is None:
+                return {
+                           'message': 'Offer_Contract_Relationship not found'
+                       }, 404
+            return ocr_with_id.to_dict()
+
+        marketplace_offer_id = request.args.get('marketplace_offer_id')
+        contract_id = request.args.get('contract_id')
+        status = request.args.get('status')
+
+        possible_filters = {
+            'marketplace_offer_id': marketplace_offer_id,
+            'contract_id': contract_id,
+            'status': status
+        }
+        filters = {}
+        for key, value in possible_filters.items():
+            if value is not None:
+                filters[key] = value
+
+        if len(filters) == 0:
             return [x.to_dict() for x in
                     ocr.OfferContractRelationship.get_all(g.context)]
-        ocrs = ocr.OfferContractRelationship.get(
-            g.context,
-            contract_id=contract_id,
-            marketplace_offer_id=marketplace_offer_id)
 
+        ocrs = ocr.OfferContractRelationship.get_all(g.context, **filters)
```

Comment:
I don't think we need to separate the cases where there are filters versus when there are no filters; if the latter case, passing in an empty filter dict should still just work, correct?

Reply:
yep, that's correct, removed